src/librustc_mir/borrow_check/mod.rs

```diff
@@ -341,7 +341,7 @@ fn do_mir_borrowck<'a, 'gcx, 'tcx>(
     }
 
     if !mbcx.errors_buffer.is_empty() {
-        mbcx.errors_buffer.sort_by_key(|diag| diag.span.primary_span());
+        mbcx.errors_buffer.sort_by_cached_key(|diag| diag.span.primary_span());
```

Comment:
This is the 'error reporting' path, so it really doesn't matter... also `primary_span` is a very simple operation... doesn't strike me as worthwhile.